internal/providers/terraform/azure/firewall.go

```diff
@@ -24,6 +24,14 @@ func NewAzureFirewall(d *schema.ResourceData, u *schema.UsageData) *schema.Resou
 		skuTier = d.Get("sku_tier").String()
 	}
 
+	if v := d.Get("virtual_hub").String(); v != "[]" {
```

Comment:
I think add your explanation from the pr as a code comment here.  "I compare d.Get() with empty array because defaulty exists empty array of virtual hub block: "virtual_hub":[] and it means that d.Get("virtual_hub").Type will never return gjson.Null"